src/components/schedule-visualizer/robot-image-icon.tsx

```diff
@@ -0,0 +1,63 @@
+import { makeStyles } from '@material-ui/core';
+import React, { useMemo } from 'react';
+import { transformMiddleCoordsOfRectToSVGBeginPoint } from '../../util/calculation-helpers';
+import { RobotProps } from './robot';
+
+const useStyles = makeStyles(() => ({
+  robotImg: {
+    transformOrigin: 'center',
+  },
+  robotImgContainer: {
+    pointerEvents: 'visible',
+  },
+}));
+
+type RobotImageIconProps = Omit<RobotProps, 'colorManager' | 'fleetName'> & {
+  iconPath: string;
+  dispatchIconError: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const RobotImageIcon = React.forwardRef(function(
+  props: RobotImageIconProps,
+  ref: React.Ref<SVGGElement>,
+): React.ReactElement {
+  const classes = useStyles();
+  const { robot, footprint, iconPath, dispatchIconError, onClick } = props;
+  // Hardcoded delta of 2 to set the size of the robot. With the double size of the footprint, we achieved a similar size to the robot default svg icon.
+  const [imgIconWidth, imgIconHeigth] = useMemo(() => [footprint * 2, footprint * 2], [footprint]);
```

Comment:
The default icon uses footprint as the radius, so we * 2 here because the width/height is in a square, or diameter. Might want to describe this instead of saying it is hardcoded.

Reply:
Makes sense. Thank you.